flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/conversion/HiveInspectors.java

```diff
@@ -244,7 +247,11 @@ public static Object toFlinkObject(ObjectInspector inspector, Object data) {
 					inspector instanceof BinaryObjectInspector) {
 
 				PrimitiveObjectInspector poi = (PrimitiveObjectInspector) inspector;
-				return poi.getPrimitiveJavaObject(data);
+				Object javaObject = poi.getPrimitiveJavaObject(data);
+				if (inspector instanceof TimestampObjectInspector) {
```

Comment:
shall we move `inspector instanceof TimestampObjectInspector` to its own else-if? rather than doing it twice here

Reply:
+1